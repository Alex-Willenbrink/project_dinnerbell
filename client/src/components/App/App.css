.App {
  text-align: center;
  background-color: #3c8d41;
}
