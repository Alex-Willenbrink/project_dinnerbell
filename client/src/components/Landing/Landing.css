@import url('https://fonts.googleapis.com/css?family=Indie+Flower|Open+Sans:400,700');

.landing {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100vw;
  height: 100vh;
  background-color: #3c8d41;
  position: relative;
}

.landing-images {
  width: 315px;
  margin-bottom: 7px;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.landing-title {
  font-family: 'Indie Flower', cursive;
  font-size: 5em;
  color: white;
}

.landing-description {
  font-family: 'Open Sans', sans-serif;
  color: white;
  position: absolute;
  top: 21%;
  z-index: 3;
}

.landing-description span {
  font-weight: bold;
}

.landing-img {
  width: 100%;
  object-fit: fill;
}

.landing-auth-container {
  display: flex;
  width: 200px;
  justify-content: space-between;
  top: 32%;
  position: absolute;
}

.landing-auth-container button {
  font-family: 'Open Sans', sans-serif;
  font-size: 16px;
  padding: 5px;
  padding-right: 10px;
  padding-left: 10px;
  border: 0.5px solid white;
  color: white;
  background-color: #3c8d41;
  border-radius: 5px;
}

/* Row 1 */

.landing-img-container-1 {
  width: 100px;
  height: 100px;
  background-image: url('https://www.bellamysorganic.com.au/blog/wp-content/uploads/2013/12/understanding-the-health-benefits-of-organic-food-1.jpg');
  background-size: cover;
  background-repeat: no-repeat;
  background-position: 50% 50%;
}

.landing-img-container-2 {
  width: 100px;
  height: 100px;
  background-image: url('http://cdn.blessthisstuff.com/imagens/stuff/sprig-organic-meals-3.jpg');
  background-size: cover;
  background-repeat: no-repeat;
  background-position: 50% 50%;
}

.landing-img-container-3 {
  width: 100px;
  height: 100px;
  background-image: url('http://www.singletonsseafoodshack.com/wp-content/uploads/2017/05/Restaurant-Eating.jpg');
  background-size: cover;
  background-repeat: no-repeat;
  background-position: 50% 50%;
}

/* Row 2 */

.landing-img-container-4 {
  width: 100px;
  height: 100px;
  background-image: url('http://roguehealthandfitness.com/wp-content/uploads/2015/02/organic.jpg');
  background-size: cover;
  background-repeat: no-repeat;
  background-position: 50% 50%;
}

.landing-img-container-5 {
  width: 100px;
  height: 100px;
  background-image: url('http://cdn.theatlantic.com/static/mt/assets/food/salad.jpg');
  background-size: cover;
  background-repeat: no-repeat;
  background-position: 50% 50%;
}

.landing-img-container-6 {
  width: 100px;
  height: 100px;
  background-image: url('http://readynutrition.com/wp-content/uploads/2014/04/shutterstock_151441463.jpg');
  background-size: cover;
  background-repeat: no-repeat;
  background-position: 50% 50%;
}

/* Row 3 */

.landing-img-container-7 {
  width: 100px;
  height: 100px;
  background-image: url('http://2zcb9n15d8i73ncvp0lwwrct.wpengine.netdna-cdn.com/wp-content/uploads/organic-whole-grain-oatmeal-pancakes-768x423.png');
  background-size: cover;
  background-repeat: no-repeat;
  background-position: 50% 50%;
}

.landing-img-container-8 {
  width: 100px;
  height: 100px;
  background-image: url('https://sunbasket.com/resources/img/home/how-it-works/three-recipe-bags-2.jpg');
  background-size: cover;
  background-repeat: no-repeat;
  background-position: 50% 50%;
}

.landing-img-container-9 {
  width: 100px;
  height: 100px;
  background-image: url('https://www.rd.com/wp-content/uploads/sites/2/2016/06/10-things-know-organic-food-farming.jpg');
  background-size: cover;
  background-repeat: no-repeat;
  background-position: 50% 50%;
}

.landing-img-cube {
  margin-top: 30px;
}
