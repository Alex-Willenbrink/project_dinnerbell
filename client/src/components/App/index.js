import React, { Component } from "react";
import { BrowserRouter as Router, Route } from "react-router-dom";

import MuiThemeProvider from "material-ui/styles/MuiThemeProvider";

import Landing from "../Landing";
import Navbar from "../Navbar";
import Authentication from "../Authentication";
import Recipes from "../Recipes";
import Profile from "../Profile";
import RecipeContainer from "../RecipeContainer";
import UsersContainer from "../UsersContainer/UsersContainer";
import Test from "../Test";
import CreateMeal from "../CreateMeal";
import MealContainer from "../MealContainer/MealContainer";

import "./App.css";

class App extends Component {
  render() {
    return (
      <MuiThemeProvider>
        <Router>
          <div style={{ height: '100%' }}>
            <Navbar />
            <Route exact path="/" render={() => <Landing />} />
            <Route
              path="/register"
              render={() => <Authentication showLogin={false} />}
            />
            <Route
              path="/login"
              render={() => <Authentication showLogin={true} />}
            />
            <Route exact path="/recipes" render={() => <Recipes {...this.props} />} />
            <Route
              path="/profile/:name"
              render={() => <Profile />}
            />
            <Route
              path="/create_meal"
              render={() => <CreateMeal />}
            />
            <Route path="/meals/:id" component={MealContainer} />
            <Route
              path="/users"
              render={() =>
                <UsersContainer>
                  <Test />
                </UsersContainer>}
            />
            <Route path="/recipes/:id" component={RecipeContainer} />
            <footer style={{ backgroundColor: '#408c45'}}></footer>
          </div>
        </Router>
      </MuiThemeProvider>
    );
  }
}

export default App;
