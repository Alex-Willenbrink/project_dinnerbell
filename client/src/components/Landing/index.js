import React, { Component } from 'react';
import { Paper } from 'material-ui';

import './Landing.css';

const style = {
  height: 100,
  width: 100
};

const Landing = () => {
  return (
    <section className="landing">
      <h1 className="landing-title">dinnerbell</h1>
      <div className="landing-auth-container">
        <button>register</button>
        <button>login</button>
      </div>
      <div className="landing-images ">
        <Paper style={style} zDepth={2}>
          <div className="landing-img-container-1" />
        </Paper>
        <Paper style={style} zDepth={2}>
          <div className="landing-img-container-2" />
        </Paper>
        <Paper style={style} zDepth={2}>
          <div className="landing-img-container-3" />
        </Paper>
      </div>
      <div className="landing-images">
        <Paper style={style} zDepth={2}>
          <div className="landing-img-container-4" />
        </Paper>
        <Paper style={style} zDepth={2}>
          <div className="landing-img-container-5" />
        </Paper>
        <Paper style={style} zDepth={2}>
          <div className="landing-img-container-6" />
        </Paper>
      </div>
      <div className="landing-images">
        <Paper style={style} zDepth={2}>
          <div className="landing-img-container-7" />
        </Paper>
        <Paper style={style} zDepth={2}>
          <div className="landing-img-container-8" />
        </Paper>
        <Paper style={style} zDepth={2}>
          <div className="landing-img-container-9" />
        </Paper>
      </div>
    </section>
  );
};

export default Landing;

// <img
//   className="landing-img"
//   src="https://www.bellamysorganic.com.au/blog/wp-content/uploads/2013/12/understanding-the-health-benefits-of-organic-food-1.jpg"
// />
