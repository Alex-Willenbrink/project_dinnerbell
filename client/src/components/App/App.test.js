import React from "react";
import App from "./index";
import Landing from "../Landing";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
configure({ adapter: new Adapter() });

describe(App, () => {
  const component = shallow(<App />);
  it("renders and matches our snapshot", () => {
    const component = renderer.create(<App />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("contains an Landing subcomponent", () => {
    expect(component.find(Landing)).toHaveLength(1);
  });
});
