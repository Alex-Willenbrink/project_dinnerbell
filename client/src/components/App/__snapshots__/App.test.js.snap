// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class App extends _react.Component {
  render() {
    return (
      _react2.default.createElement(_MuiThemeProvider2.default, { __source: { fileName: _jsxFileName, lineNumber: 10 }, __self: this },
        _react2.default.createElement(_Landing2.default, { __source: { fileName: _jsxFileName, lineNumber: 11 }, __self: this })));


  }} renders and matches our snapshot 1`] = `
<section
  className="landing"
>
  <h1
    className="landing-title"
  >
    Dinnerbell
  </h1>
  <div
    className="landing-images"
  >
    <div
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "backgroundColor": "#ffffff",
          "borderRadius": 2,
          "boxShadow": "0 3px 10px rgba(0, 0, 0, 0.16),
                 0 3px 10px rgba(0, 0, 0, 0.23)",
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "fontFamily": "Roboto, sans-serif",
          "height": 100,
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "width": 100,
        }
      }
    >
      <div
        className="landing-img-container-1"
      />
    </div>
    <div
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "backgroundColor": "#ffffff",
          "borderRadius": 2,
          "boxShadow": "0 3px 10px rgba(0, 0, 0, 0.16),
                 0 3px 10px rgba(0, 0, 0, 0.23)",
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "fontFamily": "Roboto, sans-serif",
          "height": 100,
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "width": 100,
        }
      }
    >
      <div
        className="landing-img-container-2"
      />
    </div>
    <div
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "backgroundColor": "#ffffff",
          "borderRadius": 2,
          "boxShadow": "0 3px 10px rgba(0, 0, 0, 0.16),
                 0 3px 10px rgba(0, 0, 0, 0.23)",
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "fontFamily": "Roboto, sans-serif",
          "height": 100,
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "width": 100,
        }
      }
    >
      <div
        className="landing-img-container-3"
      />
    </div>
  </div>
  <div
    className="landing-images"
  >
    <div
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "backgroundColor": "#ffffff",
          "borderRadius": 2,
          "boxShadow": "0 3px 10px rgba(0, 0, 0, 0.16),
                 0 3px 10px rgba(0, 0, 0, 0.23)",
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "fontFamily": "Roboto, sans-serif",
          "height": 100,
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "width": 100,
        }
      }
    >
      <div
        className="landing-img-container-4"
      />
    </div>
    <div
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "backgroundColor": "#ffffff",
          "borderRadius": 2,
          "boxShadow": "0 3px 10px rgba(0, 0, 0, 0.16),
                 0 3px 10px rgba(0, 0, 0, 0.23)",
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "fontFamily": "Roboto, sans-serif",
          "height": 100,
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "width": 100,
        }
      }
    >
      <div
        className="landing-img-container-5"
      />
    </div>
    <div
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "backgroundColor": "#ffffff",
          "borderRadius": 2,
          "boxShadow": "0 3px 10px rgba(0, 0, 0, 0.16),
                 0 3px 10px rgba(0, 0, 0, 0.23)",
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "fontFamily": "Roboto, sans-serif",
          "height": 100,
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "width": 100,
        }
      }
    >
      <div
        className="landing-img-container-6"
      />
    </div>
  </div>
  <div
    className="landing-images"
  >
    <div
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "backgroundColor": "#ffffff",
          "borderRadius": 2,
          "boxShadow": "0 3px 10px rgba(0, 0, 0, 0.16),
                 0 3px 10px rgba(0, 0, 0, 0.23)",
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "fontFamily": "Roboto, sans-serif",
          "height": 100,
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "width": 100,
        }
      }
    >
      <div
        className="landing-img-container-7"
      />
    </div>
    <div
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "backgroundColor": "#ffffff",
          "borderRadius": 2,
          "boxShadow": "0 3px 10px rgba(0, 0, 0, 0.16),
                 0 3px 10px rgba(0, 0, 0, 0.23)",
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "fontFamily": "Roboto, sans-serif",
          "height": 100,
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "width": 100,
        }
      }
    >
      <div
        className="landing-img-container-8"
      />
    </div>
    <div
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "backgroundColor": "#ffffff",
          "borderRadius": 2,
          "boxShadow": "0 3px 10px rgba(0, 0, 0, 0.16),
                 0 3px 10px rgba(0, 0, 0, 0.23)",
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "fontFamily": "Roboto, sans-serif",
          "height": 100,
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "width": 100,
        }
      }
    >
      <div
        className="landing-img-container-9"
      />
    </div>
  </div>
</section>
`;
